diff --git a/android/CMakeLists.txt b/android/CMakeLists.txt
index 0000000..0000000 100644
--- a/android/CMakeLists.txt
+++ b/android/CMakeLists.txt
@@ -41,14 +41,16 @@
 if(${RNS_NEW_ARCH_ENABLED})
     find_package(fbjni REQUIRED CONFIG)
     target_link_libraries(rnscreens
         ReactAndroid::reactnative
         ReactAndroid::jsi
         fbjni::fbjni
         android
+        c++_shared
     )
 else()
     target_link_libraries(rnscreens
         ReactAndroid::jsi
         android
+        c++_shared
     )
 endif()
